crates/bevy_render/src/camera/projection.rs

```diff
@@ -134,54 +134,71 @@ impl Default for PerspectiveProjection {
-#[reflect(Serialize, Deserialize)]
-pub enum WindowOrigin {
-    Center,
-    BottomLeft,
-}
-
 #[derive(Debug, Clone, Reflect, FromReflect, Serialize, Deserialize)]
 #[reflect(Serialize, Deserialize)]
 pub enum ScalingMode {
-    /// Manually specify left/right/top/bottom values.
+    /// Manually specify the projection's size with `width` and `height`.
     /// Ignore window resizing; the image will stretch.
-    None,
-    /// Match the window size. 1 world unit = 1 pixel.
-    WindowSize,
-    /// Use minimal possible viewport size while keeping the aspect ratio.
+    None { width: f32, height: f32 },
+    /// Match the window size.
+    /// The argument is the number of pixels that equals one world unit.
+    WindowSize(f32),
+    /// Use the minimal possible projection size while keeping the aspect ratio.
     /// Arguments are in world units.
     Auto { min_width: f32, min_height: f32 },
-    /// Keep vertical axis constant; resize horizontal with aspect ratio.
-    /// The argument is the desired height of the viewport in world units.
+    /// Keep the projection's height constant; adjust width to match aspect ratio.
+    /// The argument is the desired height of the projection in world units.
     FixedVertical(f32),
-    /// Keep horizontal axis constant; resize vertical with aspect ratio.
-    /// The argument is the desired width of the viewport in world units.
+    /// Keep the projection's width constant; adjust height to match aspect ratio.
+    /// The argument is the desired width of the projection in world units.
     FixedHorizontal(f32),
 }
 
+/// Project a 3D space onto a 2D surface using parallel lines, i.e., unlike [`PerspectiveProjection`],
+/// the size at which objects appear remain the same regardless of depth.
+///
+/// The volume contained in the projection is called the *view frustum*. Since the viewport is rectangular,
+/// the view frustum is in the shape of a rectangular prism.
+///
+/// Note that the cross sectional area of the view frustum and the apparent size of objects are inversely proportional.
 #[derive(Component, Debug, Clone, Reflect, FromReflect)]
 #[reflect(Component, Default)]
 pub struct OrthographicProjection {
-    pub left: f32,
-    pub right: f32,
-    pub bottom: f32,
-    pub top: f32,
+    /// The distance of the near clipping plane in world units.
+    ///
+    /// Objects closer than this will not be rendered.
     pub near: f32,
+    /// The distance of the far clipping plane in world units.
+    ///
+    /// Objects further than this will not be rendered.
     pub far: f32,
-    pub window_origin: WindowOrigin,
+    /// Specifies the origin of the viewport as a fraction of its width and height (from the bottom left corner).
+    /// Consequently, this is the point from where scaling caused by viewport resizing will occur.
+    ///
+    /// For example, if `viewport_origin` is set to (0.2, 0.6) and the size of the viewport is (10, 10),
```

Comment:
This example is not clear to me :(

Reply:
Yeah, it's not a great example. I'm still working on improving this and looking for suggestions.